mchf-eclipse/Makefile

```diff
@@ -308,13 +314,16 @@ $(BOOTLOADER):  $(BOOTLOADER).bin $(BOOTLOADER).dfu
 # compilation
 $(FIRMWARE).elf:  $(HAL_OBJS) $(DSPLIB_OBJS) $(OBJS)
 	$(ECHO) "  [LD] $@"
-	@$(CXX) $(LDFLAGS)  -Xlinker --gc-sections -Llibs -Wl,-Map,$(FIRMWARE).map -o$@ $^ $(LIBS)
+	@$(file > $(ROOTLOC)/firmware_obj_list.lst,$^)
+	@$(CXX) $(LDFLAGS)  -Xlinker --gc-sections -Llibs -Wl,-Map,$(FIRMWARE).map -o$@ @$(ROOTLOC)/firmware_obj_list.lst $(LIBS)
+	$(RM) $(call FixPath,firmware_obj_list.lst)
 
 # compilation
 $(BOOTLOADER).elf:  $(BL_HAL_OBJS) $(BL_OBJS)
 	$(ECHO) "  [LD] $@"
-	$(CXX) $(LDFLAGS) -Xlinker --gc-sections -Llibs -Wl,-Map,$(BOOTLOADER).map -o$@ $^ $(LIBS)
-
+	@$(file > $(ROOTLOC)/bootloader_obj_list.lst,$^)
+	$(CXX) $(LDFLAGS) -Xlinker --gc-sections -Llibs -Wl,-Map,$(BOOTLOADER).map -o$@ @$(ROOTLOC)/bootloader_obj_list.lst $(LIBS)
```

Comment:
Hi, I tried this construct locally and it worked for me. I have no idea yet why it does not work in the travis environment. Nevertheless we should make it work there. In any case, leave out the $(ROOTLOC) and the path issue I was mentioning should be resolved. If this also fixes the travis issue, I have no idea